python/nnvm/frontend/keras.py

```diff
@@ -345,11 +346,10 @@ def _convert_concat(insym, keras_layer, _):
 
 
 def _convert_reshape(insym, keras_layer, _):
-    shape = keras_layer.shape if hasattr(keras_layer, 'shape') \
-       else keras_layer.target_shape if hasattr(keras_layer, 'target_shape') \
-       else None
-    if shape is None:
-        raise TypeError("No shape attribute in reshape layer: {}".format(keras_layer))
+    # XXX: This works only when the target shape is in `channels_last` layout.
+    logging.warning('Assumes that the target shape is in `channels_last` layout.')
+    ch = keras_layer.target_shape[-1]
+    shape = (-1, ch) + keras_layer.target_shape[:-1]
```

Comment:
seems here the logic should be:
ch = keras_layer.input.shape[-1].value
assert ch == keras_layer.target_shape[-1], "only supports last dimension in target shape being equal to the channel number of input tensor"
shape = (-1, ch) + keras_layer.target_shape[:-1]

@kazum Can you verify it?

Reply:
Thanks, it looks good to me.